lib/spack/spack/environment.py

```diff
@@ -617,7 +694,88 @@ def install(self, user_spec, concrete_spec=None, **install_args):
                 concrete = spec.concretized()
                 self._add_concrete_spec(spec, concrete)
 
-        concrete.package.do_install(**install_args)
+        self._install(concrete, **install_args)
+
+    def _install(self, spec, **install_args):
+        # Make sure log directory exists
+        log_path = self.log_path
+        fs.mkdirp(log_path)
+
+        with fs.working_dir(self.path):
+            spec.package.do_install(**install_args)
+
+            # Link the resulting log file into logs dir
+            build_log_link = os.path.join(
+                log_path, '%s-%s.log' % (spec.name, spec.dag_hash(7)))
+            if os.path.lexists(build_log_link):
+                os.remove(build_log_link)
+            os.symlink(spec.package.build_log_path, build_log_link)
+
+    def view(self):
+        return YamlFilesystemView(
+            self._view_path, spack.store.layout, ignore_conflicts=True)
+
+    def regenerate_view(self):
+        if not self._view_path:
+            tty.debug("Skip view update, this environment does not"
+                      " maintain a view")
+            return
+
+        specs_for_view = list()
+        for spec in self._get_environment_specs():
+            # The view does not store build deps, so if we want it to
+            # recognize environment specs (which do store build deps), then
+            # they need to be stripped
+            specs_for_view.append(spack.spec.Spec.from_dict(
+                spec.to_dict(all_deps=False)
+            ))
+        installed_specs_for_view = set(s for s in specs_for_view
+                                       if s.package.installed)
+
+        view = self.view()
+        view.purge_broken_links()
+        view.purge_empty_directories()
+        specs_in_view = set(view.get_all_specs())
+        tty.msg("Updating view at {0}".format(self._view_path))
+
+        rm_specs = specs_in_view - installed_specs_for_view
+        view.remove_specs(*rm_specs, with_dependents=False)
+
+        add_specs = installed_specs_for_view - specs_in_view
+        view.add_specs(*add_specs, with_dependencies=False)
+
+    def _shell_vars(self):
+        updates = [
+            ('PATH', ['bin']),
+            ('MANPATH', ['man', 'share/man']),
+            ('ACLOCAL_PATH', ['share/aclocal']),
+            ('LD_LIBRARY_PATH', ['lib', 'lib64']),
+            ('LIBRARY_PATH', ['lib', 'lib64']),
+            ('CPATH', ['include']),
+            ('PKG_CONFIG_PATH', ['lib/pkgconfig', 'lib64/pkgconfig']),
+            ('CMAKE_PREFIX_PATH', ['']),
+        ]
```

Comment:
Is there a way to consolidate this list of default vars to update with defaults for modules?  I think the lists should come from the same place.  Or is this something that could be saved for a later PR because it would need its own new config file?

Reply:
I can see value in consolidating "knowledge of important variables" but since the types of modifications differ, I'm not sure what can be saved beyond the names of the vars themselves. I'm inclined to push this discussion to another PR but I could be missing your point. More info below:

Modules get all their environment settings from `build_environment` (and whatever packages define in their `setup/_dependent/_environment` ). 

The considerations of this and modules are in at least one spot in conflict: namely this adds the view's include path to `CPATH` and `build_environment` explicitly unsets it (although packages can undo that). That was added in https://github.com/spack/spack/pull/10017/commits/15b49eb3197b5e6aad59966220ec37bd260a04db. I can agree that it would be beneficial in this context (whereas in the `build_environment` context I can see why it is unset).

Reply:
I think this is the comment we discussed in person yesterday and all agreed it shouldn't be handled as part of this PR.

Reply:
Yep -- table it for now.  

I can imagine different users wanting/not wanting `CPATH` and maybe others, similar to how they do for modules.  This is probably something that should have sensible defaults and configuration options in `spack.yaml`.